language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/TimeClientImplTest.scala

```diff
@@ -71,14 +83,73 @@ class TimeClientImplTest extends FlatSpec with Matchers with OptionValues with D
   behavior of "[9.4] TimeClientImpl.setTime"
 
   it should "return an error without sending a request when the time to set if bigger than the current time" in {
-    ledgerServices.withTimeClient(Seq(TimeServiceImpl())) { timeClient =>
-      val currentTime = Instant.ofEpochSecond(1l, 2l)
-      intercept[RuntimeException](
-        timeClient
-          .setTime(currentTime, currentTime)
-          .timeout(TestConfiguration.timeoutInSeconds, TimeUnit.SECONDS)
-          .blockingGet()
-      )
+    ledgerServices.withTimeClient(Seq(TimeServiceImpl.createWithRef(Seq.empty, authorizer)._1)) {
+      timeClient =>
+        val currentTime = Instant.ofEpochSecond(1l, 2l)
+        intercept[RuntimeException](
+          timeClient
+            .setTime(currentTime, currentTime)
+            .timeout(TestConfiguration.timeoutInSeconds, TimeUnit.SECONDS)
+            .blockingGet()
+        )
+    }
+  }
+
+  behavior of "Authorization"
+
+  def toAuthenticatedServer(fn: TimeClient => Any): Any =
+    ledgerServices.withTimeClient(
+      Seq(TimeServiceImpl.createWithRef(Seq(genGetTimeResponse), authorizer)._1),
+      mockedAuthService)(fn)
+
+  it should "deny access without a token" in {
+    withClue("getTime") {
+      expectPermissionDenied {
+        toAuthenticatedServer {
+          _.getTime().blockingFirst()
+        }
+      }
+    }
+    withClue("setTime") {
+      expectPermissionDenied {
+        toAuthenticatedServer { client =>
+          val t = Instant.now()
+          client.setTime(t, t.plusSeconds(1)).blockingGet()
+        }
+      }
+    }
+  }
+
+  it should "deny access with insufficient authorization" in {
+    withClue("getTime") {
+      expectPermissionDenied {
+        toAuthenticatedServer {
+          _.getTime(emptyToken).blockingFirst()
```

Comment:
```suggestion
          _.getTime(somePartyReadWriteToken).blockingFirst()
```
Using `emptyToken` is better for checking whether empty tokens are handled correctly, using `somePartyReadWriteToken` checks whether this endpoint is admin-only (which it should be).

Reply:
`getTime` is public :wink: 
```scala
  override def getTime(
      request: GetTimeRequest,
      responseObserver: StreamObserver[GetTimeResponse]): Unit =
    authorizer.requirePublicClaimsOnStream(service.getTime)(request, responseObserver)
```

Reply:
Disregard the above, the `getTime` call requires public claims only. The information at https://github.com/digital-asset/daml/blob/master/ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/Claims.scala#L71 should be updated.

Reply:
I've addressed it in this PR https://github.com/digital-asset/daml/pull/3727